### ISSUE TITLE
None cannot be assigned to typing.Hashable

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#985]
**Describe the bug**
`None` cannot be assigned to `typing.Hashable`.  `None` is hashable in Python.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That `None` would be accepted as a value of type `Hashable`.

**Screenshots or Code**
```python
from typing import Dict, Hashable

# Expression of type "None" cannot be assigned to declared type "Hashable"
#   Type "None" cannot be assigned to type "Hashable"
foo: Hashable = None

# Expression of type "dict[None, str]" cannot be assigned to declared type "Dict"
#   TypeVar "_KT" is invariant
#     Type "None" cannot be assigned to type "Hashable"
bar: Dict[Hashable, str] = {None: ''}

# True
isinstance(None, Hashable)
```

**VS Code extension or command-line**
Extension, 1.1.65

**Comment by erictraut** [comment]
I'm not convinced this is a bug in pyright. Mypy also reports errors the same errors. 

Maybe the builtins.pyi typeshed stub should define a NoneType that defines the methods that the NoneType class supports.

I'll need to think about this more.

**Comment by layday** [comment]
NoneType is not in builtins however. The only way to access NoneType in Python is by calling type on None. Even then perhaps it could be stubbed out for the benefit of type checkers.

**Comment by erictraut** [comment]
I've submitted a PR to typeshed to add a `_NoneType` class to `typing.pyi`. Once that's incorporated, I can add support for this feature.

**Comment by erictraut** [comment]
The PR was accepted to typeshed, so this is now unblocked.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.74, which I just published. It will also be included in the next release of Pylance.
